yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -390,8 +390,9 @@ private[spark] class Client(
     // Upload Spark and the application JAR to the remote file system if necessary,
     // and add them as local resources to the application master.
     val fs = destDir.getFileSystem(hadoopConf)
-    val nns = YarnSparkHadoopUtil.get.getNameNodesToAccess(sparkConf) + destDir
-    YarnSparkHadoopUtil.get.obtainTokensForNamenodes(nns, hadoopConf, credentials)
+    hdfsTokenProvider(sparkConf).setNameNodesToAccess(sparkConf, Set(destDir))
```

Comment:
it seems a bit odd to me that we are doing these extra things for hdfs outside of the token provider.

what happens if a user needs to do something similar for some other service they want to implement their own class to handle?